src/backend/bacon-video-widget-gst-missing-plugins.c

```diff
@@ -304,13 +309,230 @@ bacon_video_widget_start_plugin_installation (TotemCodecInstallContext *ctx,
+	return FALSE;
+}
+
+static CodecInfo *
+codec_info_new (const gchar *codec)
+{
+	GstStructure *s;
+	CodecInfo *info = NULL;
+	g_autofree gchar *caps = NULL;
+	g_autofree gchar *type_name = NULL;
+	g_auto(GStrv) split = NULL;
+	g_auto(GStrv) ss = NULL;
+
+	split = g_strsplit (codec, "|", -1);
+	if (split == NULL || g_strv_length (split) != 5) {
+		g_warning ("Not a GStreamer codec line");
+		return NULL;
+	}
+	if (g_strcmp0 (split[0], "gstreamer") != 0) {
+		g_warning ("Not a GStreamer codec request");
+		return NULL;
+	}
+	if (g_strcmp0 (split[1], "1.0") != 0) {
+		g_warning ("Not recognised GStreamer version");
+		return NULL;
+	}
+
+	if (g_str_has_prefix (split[4], "uri") != FALSE) {
+		g_warning ("Couldn't find codec related information");
+		return NULL;
+	}
+
+	ss = g_strsplit (split[4], "-", 2);
+	type_name = g_strdup (ss[0]);
+	caps = g_strdup (ss[1]);
+
+	s = gst_structure_from_string (caps, NULL);
+	if (s == NULL) {
+		g_warning ("totem: failed to parse caps: %s", caps);
+		return NULL;
+	}
+
+	info = g_new0 (CodecInfo, 1);
+	info->type_name = g_steal_pointer (&type_name);
+	info->structure = s;
+	return info;
+}
+
+static void
+codec_info_free (CodecInfo *codec)
+{
+	gst_structure_free (codec->structure);
+	g_free (codec->type_name);
+	g_free (codec);
+}
+
+static gboolean
+missing_codecs_supported (const char **codecs)
+{
+	g_autoptr(GPtrArray) codecs_array = NULL;
+	guint len, i;
+
+	/* First of all filter the detailed strings for each missing codec
+	 * so that we only consider those we're able to successfully parse. */
+	len = g_strv_length ((GStrv)codecs);
+	codecs_array = g_ptr_array_new_with_free_func ((GDestroyNotify) codec_info_free);
+	for (i = 0; i < len; i++) {
+		CodecInfo *info = codec_info_new (codecs[i]);
+		if (info == NULL)
+			continue;
+
+		g_ptr_array_add (codecs_array, info);
+	}
+
+	/* No successfully parsed codecs strings, we can't tell */
+	if (codecs_array->len == 0)
+		return FALSE;
+
+	/* All the codecs must be supported for us to be able to declare that
+	 * we support the missing codecs with our codecs pack */
+	for (i = 0; i < codecs_array->len; i++) {
+		CodecInfo *info = g_ptr_array_index (codecs_array, i);
+
+		if (info->structure == NULL)
+			return FALSE;
+
+		/* No encoders are supported */
+		if (g_strcmp0 (info->type_name, "decoder") != 0)
+			return FALSE;
+
+		if (!is_supported_codec (info->structure))
+			return FALSE;
+	}
+
+	/* All codecs supported if reached */
+	return TRUE;
+}
+
+static void
+codec_confirmation_dialog_response_cb (GtkDialog       *dialog,
+                                       GtkResponseType  response_type,
+                                       gpointer	 user_data)
+{
+	GError *error = NULL;
+	const char * const *languages = g_get_language_names ();
+	const char *shop_uri = NULL;
+
+	switch (response_type) {
+	case GTK_RESPONSE_ACCEPT:
+		/* Different URLs for the three different shops */
+		if (g_strv_contains (languages, "es"))
```

Comment:
I had exactly the same doubt when writing this, but in the end I thought that a Spanish (from Spain) user buying codecs would normally appreciate a website in Spanish, even if having to use a different currency to his own, than a site in English which would also use a different currency anyway. And then I applied the same reasoning to Portuguese... :-)

IMHO, the ideal situation would be to have some kind of "gateway shop" that you always go to and let the server choose based on the language, I think.